yt_dlp/extractor/crunchyroll.py

```diff
@@ -827,22 +893,80 @@ def _real_extract(self, url):
         }
 
 
-class CrunchyrollBetaShowIE(CrunchyrollBaseIE):
+class CrunchyrollBetaShowIE(CrunchyrollBetaBaseIE):
     IE_NAME = 'crunchyroll:playlist:beta'
-    _VALID_URL = r'https?://beta\.crunchyroll\.com/(?P<lang>(?:\w{1,2}/)?)series/\w+/(?P<id>[\w\-]+)/?(?:\?|$)'
+    _VALID_URL = r'https?://beta\.crunchyroll\.com/(?P<lang>(?:\w{1,2}/)?)series/(?P<id>\w+)/(?P<display_id>[\w\-]*)/?(?:\?|$)'
     _TESTS = [{
         'url': 'https://beta.crunchyroll.com/series/GY19NQ2QR/Girl-Friend-BETA',
         'info_dict': {
             'id': 'girl-friend-beta',
             'title': 'Girl Friend BETA',
         },
         'playlist_mincount': 10,
+    }, {
+        'url': 'https://beta.crunchyroll.com/series/GYJQV73V6/love-chunibyo--other-delusions---heart-throb--',
+        'info_dict': {
+            'id': 'love-chunibyo-other-delusions-heart-throb-',
+            'title': 'Love, Chunibyo & Other Delusions - Heart Throb -',
+        },
+        'playlist_mincount': 10,
     }, {
         'url': 'https://beta.crunchyroll.com/it/series/GY19NQ2QR/Girl-Friend-BETA',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
-        lang, series_id = self._match_valid_url(url).group('lang', 'id')
-        return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id.lower()}',
-                               CrunchyrollShowPlaylistIE.ie_key(), series_id)
+        lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'id', 'display_id')
+
+        if not self._get_cookies(url).get('etp_rt'):
+            initial_state, app_config = self._get_beta_embedded_json(self._download_webpage(url, display_id), display_id)
+            series_id = re.sub(r'-{2,}', '-', initial_state['content']['byId'][internal_id]['slug_title'])
+            self.to_screen(f'{display_id}: Not logged in. Switching extractors. New url: https://www.crunchyroll.com/{lang}{series_id}')
+            return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id}',
+                                   CrunchyrollShowPlaylistIE.ie_key(), series_id)
+
+        api_domain, bucket, params = self._get_params(lang)
```

Comment:
repeated code, put in a func

Reply:
Honestly doesn't come out to be that much of a win. The 2 bits of code are annoyingly different.

Still, it's done.